src/services/BackendService/BackendService.ts

```diff
@@ -58,7 +60,20 @@ export class BackendService implements BackendInterface {
     };
   }
 
-  async reportDiagnosisKeys(keyPair: SubmissionKeySet, exposureKeys: TemporaryExposureKey[]) {
+  async reportDiagnosisKeys(keyPair: SubmissionKeySet, _exposureKeys: TemporaryExposureKey[]) {
+    // Ref https://github.com/CovidShield/mobile/issues/192
+    const filteredExposureKeys = Object.values(
+      _exposureKeys
+        .sort((first, second) => second.rollingStartIntervalNumber - first.rollingStartIntervalNumber)
+        .filter((value, index, array) => {
+          const previousIndex = array.findIndex(
+            lookup => lookup.rollingStartIntervalNumber === value.rollingStartIntervalNumber,
+          );
+          return previousIndex === index;
+        }),
```

Comment:
This section removes TEKs with the same rolling start number? How often does that happen?

Reply:
I encountered once in the past 2 weeks.

Reply:
Was that potentially from changing the device date? We shouldn't need to handle that particular case on the client side; let's remove this code and see if it happens again.

Reply:
Updated!